python/run-tests.py

```diff
@@ -0,0 +1,123 @@
+
+
+def test(root_dir, package):
+    # Run all of the test under test/python directory, each of them
+    # has main entry point to execute, which is python's unittest testing
+    # framework.
+    python_root_dir = path.join(root_dir, "python")
+    test_dir = path.join(python_root_dir, path.join("hyperspace", "tests"))
+    test_files = [os.path.join(test_dir, f) for f in os.listdir(test_dir)
+                  if os.path.isfile(os.path.join(test_dir, f)) and
+                  f.endswith(".py") and not f.startswith("_")]
+    extra_class_path = path.join(python_root_dir, path.join("hyperspace", "testing"))
+    for test_file in test_files:
+        try:
+            my_env = os.environ.copy()
+            my_env["SPARK_HOME"] = os.path.join(root_dir, "spark-2.4.2-bin-hadoop2.7")
+            my_env["PATH"] = os.path.join(root_dir, os.path.join("spark-2.4.2-bin-hadoop2.7", "bin")) + ":" + my_env["PATH"]
+            cmd = ["spark-2.4.2-bin-hadoop2.7/bin/spark-submit",
+                   "--driver-class-path=%s" % extra_class_path,
+                   "--jars=%s" % extra_class_path,
+                   "--packages", package, test_file]
+            print("Running tests in %s\n=============" % test_file)
+            run_cmd(cmd, stream_output=True, env=my_env)
+        except:
+            print("Failed tests in %s" % (test_file))
+            raise
+
+
+def delete_if_exists(path):
+    # if path exists, delete it.
+    if os.path.exists(path):
+        shutil.rmtree(path)
+        print("Deleted %s " % path)
+
+
+def prepare(root_dir):
+    sbt_path = "sbt"
+    delete_if_exists(os.path.expanduser("~/.ivy2/cache/com/microsoft/hyperspace"))
+    delete_if_exists(os.path.expanduser("~/.m2/repository/com/microsoft/hyperspace"))
+    run_cmd([sbt_path, "clean", "publishM2"], stream_output=True)
+    package = "com.microsoft.hyperspace:hyperspace-core_2.12:0.2.0-SNAPSHOT"
+    return package
+
+
+def run_cmd(cmd, throw_on_error=True, env=None, stream_output=False, **kwargs):
+    cmd_env = os.environ.copy()
+    if env:
+        cmd_env.update(env)
+
+    if stream_output:
+        child = subprocess.Popen(cmd, env=cmd_env, **kwargs)
+        exit_code = child.wait()
+        if throw_on_error and exit_code != 0:
+            raise Exception("Non-zero exitcode: %s" % (exit_code))
+        return exit_code
+    else:
+        child = subprocess.Popen(
+            cmd,
+            env=cmd_env,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            **kwargs)
+        (stdout, stderr) = child.communicate()
+        exit_code = child.wait()
+        if throw_on_error and exit_code is not 0:
+            raise Exception(
+                "Non-zero exitcode: %s\n\nSTDOUT:\n%s\n\nSTDERR:%s" %
+                (exit_code, stdout, stderr))
+        return (exit_code, stdout, stderr)
+
+
+def run_python_style_checks(root_dir):
+    run_cmd([os.path.join(root_dir, "dev", "lint-python")], stream_output=True)
+
+
```

Comment:
this file has a different style?

Reply:
Actually haven't integrated this lint check in this PR? Can do it in separate PR?